diff --git a/pkg/utils/utils.go b/pkg/utils/utils.go
index 5f480509..98480735 100644
--- a/pkg/utils/utils.go
+++ b/pkg/utils/utils.go
@@ -70,12 +70,33 @@ func GetPfName(pciAddr string) (string, error) {
 					path := filepath.Join(sysBusPci, pciAddr, "physfn/virtio0/net")
 					files, err = ioutil.ReadDir(path)
 					if err != nil {
+						if os.IsNotExist(err) {
+							// Read /sys/bus/pci/devices/<PCIAddr>/
+							path := filepath.Join(sysBusPci, pciAddr)
+							files, err = ioutil.ReadDir(path)
+							if err != nil {
+								return "", err
+							}
+							// Search for directory "virtioX", where x is usually a number 0-x
+							for _, file := range files {
+								if match, _ := regexp.MatchString("virtio*", file.Name()); match == true {
+									path := filepath.Join(sysBusPci, pciAddr, file.Name(), "net")
+									files, err = ioutil.ReadDir(path)
+									if err != nil {
+										return "", err
+									}
+									if len(files) < 1 {
+										return "", fmt.Errorf("no interface name found for device %s", pciAddr)
+									}
+									return files[0].Name(), nil
+								}
+							}
+							return "", err
+						}
 						return "", err
+					} else if len(files) > 0 {
+						return files[0].Name(), nil
 					}
-					if len(files) < 1 {
-						return "", fmt.Errorf("no interface name found for device %s", pciAddr)
-					}
-					return files[0].Name(), nil
 				}
 				return "", err
 			} else if len(files) > 0 {
